Was able to run locally and get error:

```
2024-09-23 14:09:06,409 INFO  [org.hibernate.cache.internal.RegionFactoryInitiator] (ServerService Thread Pool -- 13) HHH000026: Second-level cache disabled
2024-09-23 14:09:06,440 INFO  [org.hibernate.envers.boot.internal.EnversServiceImpl] (ServerService Thread Pool -- 13) Envers integration enabled? : true
2024-09-23 14:09:06,523 INFO  [org.jboss.as.jpa] (ServerService Thread Pool -- 13) WFLYJPA0010: Starting Persistence Unit (phase 2 of 2) Service 'simple-app-client-test.ear#mainPu'
2024-09-23 14:09:06,552 INFO  [org.hibernate.orm.beans] (ServerService Thread Pool -- 13) HHH10005002: No explicit CDI BeanManager reference was passed to Hibernate, but CDI is available on the Hibernate ClassLoader.
2024-09-23 14:09:06,687 INFO  [org.hibernate.validator.internal.util.Version] (ServerService Thread Pool -- 13) HV000001: Hibernate Validator 8.0.1.Final
2024-09-23 14:09:06,968 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly 34.0.0.Beta1-SNAPSHOT (WildFly Core 26.0.0.Beta5) started in 1855ms - Started 249 of 257 services (22 services are lazy, passive or on-demand) - Server configuration file in use: appclient.xml - Minimum feature stability level: community
2024-09-23 14:09:06,970 INFO  [org.jboss.as.server] (Thread-44) WFLYSRV0010: Deployed "simple-app-client-test.ear" (runtime-name : "simple-app-client-test.ear")
2024-09-23 14:09:06,973 INFO  [org.jboss.ejb.client] (Thread-51) JBoss EJB Client version 5.0.7.Final
2024-09-23 14:09:06,977 ERROR [org.jboss.as.appclient] (Thread-51) WFLYAC0002: RuntimeException running app client main: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at org.jboss.as.appclient@34.0.0.Beta1-SNAPSHOT//org.jboss.as.appclient.service.ApplicationClientStartService$1.run(ApplicationClientStartService.java:117)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at org.jboss.as.appclient@34.0.0.Beta1-SNAPSHOT//org.jboss.as.appclient.service.ApplicationClientStartService$1.run(ApplicationClientStartService.java:89)
        ... 1 more
Caused by: java.lang.IllegalAccessError: class org.jboss.as.test.integration.ee.appclient.basic.AppClientMain tried to access method 'java.lang.String org.jboss.as.test.integration.ee.appclient.basic.Status.getStatus()' (org.jboss.as.test.integration.ee.appclient.basic.AppClientMain is in unnamed module of loader 'deployment.simple-app-client-test.ear.client-annotation.jar' @7b04a7a0; org.jboss.as.test.integration.ee.appclient.basic.Status is in unnamed module of loader 'deployment.simple-app-client-test.ear' @1b345d83)
        at deployment.simple-app-client-test.ear.client-annotation.jar//org.jboss.as.test.integration.ee.appclient.basic.AppClientMain.main(AppClientMain.java:27)
        ... 6 more


```

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AppClientMain.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AppClientMain.java
@@ -24,7 +24,8 @@ public class AppClientMain {
 
     public static void main(final String[] params) {
         logger.trace("Main method invoked");
-
+        Status status = new Status();
+        logger.info("Status class accessed via classloader = " + status.getClass().getClassLoader());
         if(!appclient) {
             logger.error("InAppClientContainer was not true");
             throw new RuntimeException("InAppClientContainer was not true");

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/SimpleApplicationClientTestCase.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/SimpleApplicationClientTestCase.java
@@ -44,6 +44,11 @@ public class SimpleApplicationClientTestCase extends AbstractSimpleApplicationCl
         lib.addClasses(Employee.class);
         lib.addAsManifestResource(PersistenceUnitPackagingTestCase.class.getPackage(), "persistence.xml", "persistence.xml");
         ear.addAsLibrary(lib);
+
+        final JavaArchive otherLib = ShrinkWrap.create(JavaArchive.class, "otherlib.jar");
+        otherLib.addClass(Status.class);
+        ear.addAsLibrary(otherLib);
+
         final JavaArchive ejb = ShrinkWrap.create(JavaArchive.class, MODULE_NAME + ".jar");
         ejb.addClasses(SimpleApplicationClientTestCase.class, AppClientStateSingleton.class);
         ear.addAsModule(ejb);

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/SimpleApplicationClientTestCase2.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/SimpleApplicationClientTestCase2.java
@@ -44,6 +44,11 @@ public class SimpleApplicationClientTestCase2 extends AbstractSimpleApplicationC
         lib.addClasses(Employee.class);
         lib.addAsManifestResource(PersistenceUnitPackagingTestCase.class.getPackage(), "persistence.xml", "persistence.xml");
         ear.addAsLibrary(lib);
+
+        final JavaArchive otherLib = ShrinkWrap.create(JavaArchive.class, "otherlib.jar");
+        otherLib.addClass(Status.class);
+        ear.addAsLibrary(otherLib);
+
         final JavaArchive ejb = ShrinkWrap.create(JavaArchive.class, MODULE_NAME + ".jar");
         ejb.addClasses(SimpleApplicationClientTestCase2.class, AppClientStateSingleton.class);
         ear.addAsModule(ejb);

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/Status.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/Status.java
@@ -1,0 +1,17 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.test.integration.ee.appclient.basic;
+
+/**
+ * Status
+ *
+ * @author Scott Marlow
+ */
+public class Status {
+    public String getStatus() {
+        return "there were no test failures accessing the Status class!";
+    }
+}

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/UnspecifiedApplicationClientTestCase.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/UnspecifiedApplicationClientTestCase.java
@@ -46,6 +46,11 @@ public class UnspecifiedApplicationClientTestCase  extends AbstractSimpleApplica
         lib.addClasses(Employee.class);
         lib.addAsManifestResource(PersistenceUnitPackagingTestCase.class.getPackage(), "persistence.xml", "persistence.xml");
         ear.addAsLibrary(lib);
+
+        final JavaArchive otherLib = ShrinkWrap.create(JavaArchive.class, "otherlib.jar");
+        otherLib.addClass(Status.class);
+        ear.addAsLibrary(otherLib);
+
         final JavaArchive ejb = ShrinkWrap.create(JavaArchive.class, MODULE_NAME + ".jar");
         ejb.addClasses(SimpleApplicationClientTestCase.class, AppClientStateSingleton.class);
         ear.addAsModule(ejb);

